plugins/toolbox-search/src/block_searcher.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly/core';
+
+/**
+ * A class that provides methods for indexing and searching blocks.
+ */
+export class BlockSearcher {
+  private blockCreationWorkspace = new Blockly.Workspace();
+  private trigramsToBlocks = new Map<string, Set<string>>();
+
+  /**
+   * Populates the cached map of trigrams to the blocks they correspond to.
+   * @param blockIds A list of block IDs to index.
+   */
+  indexBlocks(blockIds: string[]) {
+    blockIds.forEach((blockId) => {
+      const block = this.blockCreationWorkspace.newBlock(blockId);
+      this.indexBlockText(blockId.replaceAll('_', ' '), blockId);
```

Comment:
The caller verifies this; is there a way we can check their validity here/is that what you're suggesting?

Reply:
Nope, checking on the naming because I expected them to be the UUIDs. Can you change the parameter/variable names to blockType and blockTypes instead of ID?

Reply:
Done!